src/main/java/data_access/SearchHistoryDataAccessObject.java

```diff
@@ -0,0 +1,27 @@
+package data_access;
+
+import entity.SearchHistoryRecord;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Handles storing and retrieving search history records.
+ */
+public class SearchHistoryDataAccessObject {
+    private final List<SearchHistoryRecord> history = new ArrayList<>();
+
+    /**
+     * Saves a search history record.
+     */
+    public void saveSearchHistory(SearchHistoryRecord record) {
+        history.add(record);
+    }
+
+    /**
+     * Retrieves all search history records.
+     */
+    public List<SearchHistoryRecord> getSearchHistory() {
+        return new ArrayList<>(history); // Return a copy to prevent external modification
```

Comment:
Good practice to return a copy. You might add comments to specify why immutability is important in this context.